frappe/email/receive.py

```diff
@@ -795,11 +795,9 @@ def parent_communication(self):
 		if not self.is_reply():
 			return ""
 
-		if not self.is_reply_to_system_sent_mail():
-			communication = Communication.find_one_by_filters(
-				message_id=self.in_reply_to, creation=[">=", self.get_relative_dt(-30)]
```

Comment:
removed creation filter. Dont see any reason why email older than 30 days shouldn't be linked.